doc/KUKSA.val_data_broker/authorization.md

```diff
@@ -0,0 +1,561 @@
+token will then only grant this subset of authorization to the client (e.g. a third party app)
+using the token.
+
+There are currently no plans in KUKSA.VAL to separately verify the access rights of the
+resource owner itself, or what kind of access rights a certain group membership or role would
+provide. What this means in practice is that anyone authorized to get a valid access token is
+assumed to have full access rights within KUKSA.VAL, which will only be limited by the scope
+provided.
+
+There is nothing in the design preventing KUKSA.VAL from including support for other types
+of verifications in the future. The scope as specified here will continue to work as a way to
+limit the access granted by these other means anyway.
+
+
+### Limit access to resources
+The issued access token will typically be intended to for use with a certain resource. In the
+context of KUKSA.VAL, this can be a certain vehicle, or a certain backend service.
+
+To prevent access tokens issued for one resource from being used to gain access to other,
+unintended resources, OAuth2 introduces the concept of
+[Resource Indicators [RFC 8707]](https://datatracker.ietf.org/doc/html/rfc8707). By specifying
+the intended resource in the request to the authorization server, the resulting `"aud"` or
+audience claim should be populated with this resource.
+
+From [RFC9068](https://datatracker.ietf.org/doc/html/rfc9068#JWTATLRequest):
+> If the request includes a "resource" parameter (as defined in
+  [[RFC8707](https://datatracker.ietf.org/doc/html/rfc6749)]), the resulting JWT access token
+  "aud" claim SHOULD have the same value as the "resource" parameter in the request.
+
+and from [RFC8707](https://datatracker.ietf.org/doc/html/rfc6749):
+> To prevent cross-JWT confusion, authorization servers MUST use a distinct identifier as an
+  "aud" claim value to uniquely identify access tokens issued by the same issuer for distinct
+  resources. For more details on cross-JWT confusion, please refer to Section 2.8 of [RFC8725].
+
+By using a resource identifier that uniquely identifies the resource (i.e. a certain instance of KUKSA.VAL),
+and including this in the `"aud"` claim, KUKSA.VAL can verify that the access token isn't meant
+for another resource.
+
+KUKSA.VAL can be configured to accept multiple `"aud"` claims, i.e. any one of them would be
+considered valid if present in an access token. It's also possible to have multiple `"aud"` claims
+in an access token, which would then be valid for multiple resources.
+
+The uniqueness of any audience claims is a deployment decision, where the most secure option is to
+have unique audiences for each instance (for example based on the VIN of a vehicle). However,
+depending on the use case, audiences valid for a whole fleet or certain subsets/models of vehicles
+may be the only option."
+
+Example of an `"aud"` claim:
+
+```yaml
+{
+  "aud": [
+    "5GZCZ43D13S812715/kuksa.val.v1",
```

Comment:
On line 154 we use "vehicle/5GZCZ43D13S812715/kuksa.val.v1" as example. Any particular reason to have "vehicle" as prefix in one of them and not the other? If not possibly use it in both or none.

Reply:
Yes, the reason is that I changed one of them but forgot the other :)